eslint-plugin-jsx-a11y(anchor-is-valid): working incorrectly

<details>
<summary>Example</summary>

```tsx
export const shouldBeValid = (
	<>
		<a href="/some/valid/uri">Navigate to page</a>
		<a href={"/some/valid/uri"}>Navigate to page</a>
		<a href={`/some/valid/uri`}>Navigate to page</a>
		<a href="#top">Navigate to internal page location</a>
		<a href={"#top"}>Navigate to internal page location</a>
		<a href={`#top`}>Navigate to internal page location</a>
		<a href="https://github.com">github</a>
		<a href={"https://github.com"}>github</a>
		<a href={`https://github.com`}>github</a>
		<a href="#section">section</a>
		<a href={"#section"}>section</a>
		<a href={`#section`}>section</a>
	</>
)

export const shouldBeInvalid = (
	<>
		<a href="#">invalid</a>
		<a href={"#"}>invalid</a>
		<a href={`#`}>invalid</a>
	</>
)
```
</details>

<details>
<summary>Result</summary>

Oxlint: 
```console
> bunx oxlint --jsx-a11y-plugin src\bar.tsx  

  ⚠ eslint-plugin-jsx-a11y(anchor-is-valid): Use an incorrect href for the 'a' element.
   ╭─[src\bar.tsx:3:4]
 2 │     <>
 3 │         <a href="/some/valid/uri">Navigate to page</a>
   ·          ─
 4 │         <a href={"/some/valid/uri"}>Navigate to page</a>
   ╰────
  help: Provide a correct href for the `a` element.

  ⚠ eslint-plugin-jsx-a11y(anchor-is-valid): Use an incorrect href for the 'a' element.
   ╭─[src\bar.tsx:4:4]
 3 │         <a href="/some/valid/uri">Navigate to page</a>
 4 │         <a href={"/some/valid/uri"}>Navigate to page</a>
   ·          ─
 5 │         <a href={`/some/valid/uri`}>Navigate to page</a>
   ╰────
  help: Provide a correct href for the `a` element.

  ⚠ eslint-plugin-jsx-a11y(anchor-is-valid): Use an incorrect href for the 'a' element.
   ╭─[src\bar.tsx:6:4]
 5 │         <a href={`/some/valid/uri`}>Navigate to page</a>
 6 │         <a href="#top">Navigate to internal page location</a>
   ·          ─
 7 │         <a href={"#top"}>Navigate to internal page location</a>
   ╰────
  help: Provide a correct href for the `a` element.

  ⚠ eslint-plugin-jsx-a11y(anchor-is-valid): Use an incorrect href for the 'a' element.
   ╭─[src\bar.tsx:7:4]
 6 │         <a href="#top">Navigate to internal page location</a>
 7 │         <a href={"#top"}>Navigate to internal page location</a>
   ·          ─
 8 │         <a href={`#top`}>Navigate to internal page location</a>
   ╰────
  help: Provide a correct href for the `a` element.

  ⚠ eslint-plugin-jsx-a11y(anchor-is-valid): Use an incorrect href for the 'a' element.
    ╭─[src\bar.tsx:9:4]
  8 │         <a href={`#top`}>Navigate to internal page location</a>
  9 │         <a href="https://github.com">github</a>
    ·          ─
 10 │         <a href={"https://github.com"}>github</a>
    ╰────
  help: Provide a correct href for the `a` element.

  ⚠ eslint-plugin-jsx-a11y(anchor-is-valid): Use an incorrect href for the 'a' element.
    ╭─[src\bar.tsx:10:4]
  9 │         <a href="https://github.com">github</a>
 10 │         <a href={"https://github.com"}>github</a>
    ·          ─
 11 │         <a href={`https://github.com`}>github</a>
    ╰────
  help: Provide a correct href for the `a` element.

  ⚠ eslint-plugin-jsx-a11y(anchor-is-valid): Use an incorrect href for the 'a' element.
    ╭─[src\bar.tsx:12:4]
 11 │         <a href={`https://github.com`}>github</a>
 12 │         <a href="#section">section</a>
    ·          ─
 13 │         <a href={"#section"}>section</a>
    ╰────
  help: Provide a correct href for the `a` element.

  ⚠ eslint-plugin-jsx-a11y(anchor-is-valid): Use an incorrect href for the 'a' element.
    ╭─[src\bar.tsx:13:4]
 12 │         <a href="#section">section</a>
 13 │         <a href={"#section"}>section</a>
    ·          ─
 14 │         <a href={`#section`}>section</a>
    ╰────
  help: Provide a correct href for the `a` element.

  ⚠ eslint-plugin-jsx-a11y(anchor-is-valid): Use an incorrect href for the 'a' element.
    ╭─[src\bar.tsx:20:4]
 19 │     <>
 20 │         <a href="#">invalid</a>
    ·          ─
 21 │         <a href={"#"}>invalid</a>
    ╰────
  help: Provide a correct href for the `a` element.

  ⚠ eslint-plugin-jsx-a11y(anchor-is-valid): Use an incorrect href for the 'a' element.
    ╭─[src\bar.tsx:21:4]
 20 │         <a href="#">invalid</a>
 21 │         <a href={"#"}>invalid</a>
    ·          ─
 22 │         <a href={`#`}>invalid</a>
    ╰────
  help: Provide a correct href for the `a` element.

Finished in 5ms on 1 file with 121 rules using 12 threads.
Found 10 warnings and 0 errors.
```
Eslint: 
Only the last 3
```console
  20:3  error  The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid  
  21:3  error  The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid  
  22:3  error  The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid  

✖ 3 problems (3 errors, 0 warnings)
```


</details>

We can notice that the string wrapped in backquotes will only be considered a valid href, even if it's not valid (e.g. ```<a href={`#`}>invalid</a>```)

--- a/crates/oxc_linter/src/rules/jsx_a11y/anchor_is_valid.rs
+++ b/crates/oxc_linter/src/rules/jsx_a11y/anchor_is_valid.rs
@@ -18,18 +18,18 @@ use crate::{
 
 fn missing_href_attribute(span: Span) -> OxcDiagnostic {
     OxcDiagnostic::warn("Missing `href` attribute for the `a` element.")
-        .with_help("Provide an href for the `a` element.")
+        .with_help("Provide an `href` for the `a` element.")
         .with_label(span)
 }
 
 fn incorrect_href(span: Span) -> OxcDiagnostic {
-    OxcDiagnostic::warn("Use an incorrect href for the 'a' element.")
-        .with_help("Provide a correct href for the `a` element.")
+    OxcDiagnostic::warn("Use of incorrect `href` for the 'a' element.")
+        .with_help("Provide a correct `href` for the `a` element.")
         .with_label(span)
 }
 
 fn cant_be_anchor(span: Span) -> OxcDiagnostic {
-    OxcDiagnostic::warn(" The a element has `href` and `onClick`.")
+    OxcDiagnostic::warn("The `a` element has `href` and `onClick`.")
         .with_help("Use a `button` element instead of an `a` element.")
         .with_label(span)
 }

--- a/crates/oxc_linter/src/rules/jsx_a11y/anchor_is_valid.rs
+++ b/crates/oxc_linter/src/rules/jsx_a11y/anchor_is_valid.rs
@@ -148,7 +148,7 @@ impl Rule for AnchorIsValid {
                     return;
                 };
 
-                let is_empty = self.check_value_is_empty(value);
+                let is_empty = self.check_value_is_empty_or_invalid(value);
                 if is_empty {
                     if has_jsx_prop_ignore_case(&jsx_el.opening_element, "onclick").is_some() {
                         ctx.diagnostic(cant_be_anchor(get_span()));
@@ -173,7 +173,7 @@ impl Rule for AnchorIsValid {
 }
 
 impl AnchorIsValid {
-    fn check_value_is_empty(&self, value: &JSXAttributeValue) -> bool {
+    fn check_value_is_empty_or_invalid(&self, value: &JSXAttributeValue) -> bool {
         match value {
             JSXAttributeValue::Element(_) => false,
             JSXAttributeValue::StringLiteral(str_lit) => self.is_invalid_href(&str_lit.value),
@@ -181,6 +181,16 @@ impl AnchorIsValid {
                 JSXExpression::Identifier(ident) => ident.name == "undefined",
                 JSXExpression::NullLiteral(_) => true,
                 JSXExpression::StringLiteral(str_lit) => self.is_invalid_href(&str_lit.value),
+                JSXExpression::TemplateLiteral(temp_lit) => {
+                    if !temp_lit.expressions.is_empty() {
+                        return false;
+                    }
+
+                    let Some(quasi) = temp_lit.quasi() else {
+                        return false;
+                    };
+                    self.is_invalid_href(&quasi)
+                }
                 _ => false,
             },
             JSXAttributeValue::Fragment(_) => true,
@@ -196,7 +206,11 @@ impl AnchorIsValidConfig {
     }
 
     fn is_invalid_href(&self, href: &str) -> bool {
-        href.is_empty() || href == "#" || href == "javascript:void(0)" || !self.contains(href)
+        if self.contains(href) {
+            return false;
+        }
+
+        href.is_empty() || href == "#" || href == "javascript:void(0)"
     }
 
     fn contains(&self, href: &str) -> bool {
@@ -361,6 +375,22 @@ fn test() {
             None,
         ),
         (r"<a href={this} onClick={() => void 0} />", None, None),
+        (r"<a href='/some/valid/uri'>valid</a>", None, None),
+        (r"<a href={'/some/valid/uri'}>valid</a>", None, None),
+        (r"<a href={`/some/valid/uri`}>valid</a>", None, None),
+        (r"<a href='#top'>Navigate to internal page location</a>", None, None),
+        (r"<a href={'#top'}>Navigate to internal page location</a>", None, None),
+        (r"<a href={`#top`}>Navigate to internal page location</a>", None, None),
+        (r"<a href='https://github.com'>github</a>", None, None),
+        (r"<a href={'https://github.com'}>github</a>", None, None),
+        (r"<a href={`https://github.com`}>github</a>", None, None),
+        (r"<a href='#section'>section</a>", None, None),
+        (r"<a href={'#section'}>section</a>", None, None),
+        (r"<a href={`#section`}>section</a>", None, None),
+        (r"<a href={`${foo}`}>valid</a>", None, None),
+        (r"<a href={`#${foo}`}>valid</a>", None, None),
+        (r"<a href={`#${foo}/bar`}>valid</a>", None, None),
+        (r"<a href={foo + bar}>valid</a>", None, None),
         // (r#"<Anchor {...props} onClick={() => void 0} />"#, Some(serde_json::json!(components))),
         // (r#"<Anchor href='foo' onClick={() => void 0} />"#, Some(serde_json::json!(components))),
         // (r#"<Anchor href={foo} onClick={() => void 0} />"#, Some(serde_json::json!(components))),
@@ -533,6 +563,7 @@ fn test() {
         (r"<a href=' />;", None, None),
         (r"<a href='#' />", None, None),
         (r"<a href={'#'} />", None, None),
+        (r"<a href={`#`} />", None, None),
         (r"<a href='javascript:void(0)' />", None, None),
         (r"<a href={'javascript:void(0)'} />", None, None),
         (r"<a onClick={() => void 0} />", None, None),

